docs/release-notes.md

```diff
@@ -15,10 +15,12 @@ nav_order: 9
 
 ### Changes
 
+- The Dracut module now installs partprobe
```

Comment:
I'd prefer we use `blockdev --rereadpt` which is part of util-linux and more likely to be already available (e.g. FCOS doesn't ship `partprobe`).

Reply:
I tried this and it doesn't achieve the same thing because it can fail with `BLKRRPART: Device or resource busy` while `partprobe` succeeds in forcing a kernel partition re-read. If you know that you don't have this problem you could use the introduced command parameter to run blockdev instead, does this work for you?

Reply:
I guess this requires a wrapper to pass the `--rereadpt` argument. Your initrd could ship something like that as `blockdev-rereadpt` and then you configure that as partprobeCmd:
```
#!/bin/sh
exec blockdev --rereadpt "$@"
```

Reply:
> I tried this and it doesn't achieve the same thing because it can fail with `BLKRRPART: Device or resource busy` while `partprobe` succeeds in forcing a kernel partition re-read.

Not sure why it's hitting `EBUSY`. The common case would be if one of the filesystems on that disk is mounted, but since we were just able to modify the partition table, that shouldn't be the case. Possibly we're racing with udev?

And actually on that topic, would it work here to play the same trick as https://github.com/coreos/ignition/pull/1319 and emit a tagged event against the disk?

Reply:
Right, one of the filesystems is mounted and this happens when adding a new partition to the boot disk.

Reply:
Ok, so what works is `partx -u - /dev/vda` and this should have the same effect as partprobe which also adds the individual partitions to the kernel. Since partx should be available everywhere where util-linux is, should I change the PR to use partx?

Reply:
It seems that `sgdisk` doesn't do that itself while `sfdisk` does - is there a big reason to use `sgdisk`?
Anyway, since it doesn't hurt to call `partx` I would suggest to use it here and can update the PR.

Reply:
> It seems that `sgdisk` doesn't do that itself while `sfdisk` does - is there a big reason to use `sgdisk`?

I'm not sure. It seems like sfdisk didn't support GPT in the past so that may be related, but I didn't check the timelines. Regardless, changing it at this point is not worth the risk. The implementation is quite tied to `sgdisk` intricacies.

> Anyway, since it doesn't hurt to call `partx` I would suggest to use it here and can update the PR.

Before we do that, did you try out this suggestion:

> And actually on that topic, would it work here to play the same trick as https://github.com/coreos/ignition/pull/1319 and emit a tagged event against the disk?

?

That'd be my preferred solution if it works.

I know you're aware, but for posterity: the difference with `partx` (and `partprobe`) vs `blockdev --rereadpt` is that they manually parse the table and tell the kernel about it instead of letting the kernel do it (`BLKPG` vs `BLKRRPART`). While there _shouldn't_ be a perceptible difference, I'd rather we keep relying on the kernel and not make the first boot special. (And of course, it avoids another dependency.)

Reply:
`udevadm trigger --settle` does not result in the new partition showing up, maybe that wasn't clear from my previous comments